concordium-consensus/testdata/contracts/v1/iterator.wat

```diff
@@ -0,0 +1,151 @@
+;; A contract that creates a state and traverses it by using the iterator.
+;; Furthermore it is checked that while a subtree is captured by an iterator,
+;; then one cannot carry out structural changes to the ‘locked’ subtree.
```

Comment:
Can you add that the contract itself checks these things, in the sense that it will fail with a runtime error if the properties do not hold.